example/scripts/lock-stack.js

```diff
@@ -3,48 +3,133 @@
 'use strict';
 
 /**
- * Functionality for locking stacks with expiration using the Mutex provided by
- * @cumulus/common. Stacks are locked based on the deployment name.
+ * Functionality for locking stacks with expiration
  *
  * To use this, a DynamoDB table is required with the primary partition key being
  * a string - 'key' and no sort key. The table name should be set in LOCK_TABLE_NAME
  */
 
-const {
-  concurrency
-} = require('@cumulus/common');
-const aws = require('@cumulus/common/aws');
+const { aws } = require('@cumulus/common');
+
+class CumulusLockError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+class Mutex {
```

Comment:
This was pulled from common/concurrency.js and was not being used anywhere else in our code.    

TODO: refactor

Reply:
Is there anything here that extends or re-uses what was in the original `Mutex` class? If not, this is a totally different class implementation, so I would rename this one to `DeploymentMutex` and leave the original `Mutex` class intact in the `common` package.

I do feel strongly that a class implementation should go in its own file, so here `DeploymentMutex.js` which could be a sibling to this file. Unit tests I don't feel strongly about per se, but it seems like based on the new rules I should ask for them? Particularly given how important the behavior of this class is to our CI reliability?

Reply:
Extending the class didn't feel like it makes a ton of sense given the structure and the changes.    Moving the class another file is 👍 , although I could probably just get rid of the class in favor of a more functional approach and avoid adding more structure to '/scripts'.   🤔   

> Unit tests I don't feel strongly about per se, but it seems like based on the new rules I should ask for them? Particularly given how important the behavior of this class is to our CI reliability?

I suppose there's a philosophy question here - we're also not testing our CI scripting on the whole (example doesn't run unit tests of any sort) ... outside that the tests run)   Perhaps that's  an indication that this 'script' needs it's own package, etc?   I really don't think it should be in common without a use case outside our CI.